controller/src/main/java/io/pravega/controller/store/PravegaTablesStoreHelper.java

```diff
@@ -244,6 +248,101 @@ public void invalidateCache(String table, String key) {
                 .whenComplete((r, ex) -> releaseEntries(entries));
     }
 
+    /**
+     * Remove value of every entry for the provided keys.
+     *
+     * @param tableName      Table name.
+     * @param tableKeys      List of keys whose value needs to be updated.
+     * @param removeValue    Value to be removed from all the keys.
+     * @param updateFunction Update function which is used to compute the updated entries. This returns the list of
+     *                       updated entries along with the index of entries which are empty and can be deleted.
+     * @param requestId      Request id.
+     * @return Future which completes when the operation is complete.
+     */
+    public CompletableFuture<Void> removeValues(final String tableName, Set<String> tableKeys, final String removeValue,
+                                                BiFunction<String, List<TableSegmentEntry>, Pair<List<TableSegmentEntry>, List<Integer>>> updateFunction,
+                                                long requestId) {
+        Supplier<String> errorMessage = () -> String.format("remove values: on table: %s", tableName);
+
+        // read values and remove values
+        return withRetries(() -> {
+            // Get un-versioned keys.
+            final List<TableSegmentKey> keys = tableKeys.stream()
+                                                        .map(k -> TableSegmentKey.unversioned(k.getBytes(StandardCharsets.UTF_8)))
+                                                        .collect(Collectors.toList());
+            // read entries from table.
+            return segmentHelper.readTable(tableName, keys, authToken.get(), requestId) // read the latest version of keys.
+                                .whenComplete((v, t) -> releaseKeys(keys))
+                                .thenCompose(currentEntries -> {
+                                    // Compute the updated values.
+                                    Pair<List<TableSegmentEntry>, List<Integer>> result = updateFunction.apply(removeValue, currentEntries);
+
+                                    return segmentHelper.updateTableEntries(tableName, result.getLeft(), authToken.get(), requestId)
+                                                        // Delete TagRecords with empty streams.
+                                                        .thenCompose(keyVersions -> deleteEmptyTagRecords(tableName, requestId, result.getRight(), result.getLeft(), keyVersions))
+                                                        .whenComplete((v, ex) -> {
+                                                            releaseEntries(result.getLeft());
+                                                        });
+                                });
+        }, errorMessage, true, requestId);
+    }
+
+    private CompletableFuture<Void> deleteEmptyTagRecords(String tableName, long requestId, List<Integer> deleteIndex,
```

Comment:
i thought we decided to remove this from here.

Reply:
Fixed it.